pixel-perfect breakpoint icons

jlongster opened the pull request:
Breakpoints before looked a little off and it was driving me crazy. This makes them align perfectly with the line numbers, and also load instantly. 

On my computer, I'm still having the issue where line numbers are slightly off, so the breakpoints are 1px off for me but they should be perfectly aligned for everyone else. I still need to figure out the line number issue that I'm seeing.

jlongster commented on public/js/components/Editor.js:
```diff
@@ -77,6 +81,13 @@ const Editor = React.createClass({
     }
   },
 
+  resizeBreakpointGutter() {
```
The technique here anchors the breakpoint to the beginning of the line, or the "right side" of the line number gutter. That way if there are 10,000,000 lines and the gutter expands really large, the breakpoint will be anchored to the right side and will travel with it. Unfortunately I need to do a little manual layout for that.

jasonLaster replied:
This code seems like it should live in `util/editor` as it is CM specific and does not deal with the business logic of debugging per-se

jlongster replied:
That's fine with me

jasonLaster commented on public/js/components/EditorBreakpoint.js:
```diff
@@ -1,20 +1,23 @@
+  marker.appendChild(svg);
+
   return marker;
 }
```
nice

jlongster commented:
I asked Helen to look at this too, so let's not merge until she's had a look

helenvholmes commented:
Here's a huge debugger breakpoint (hopefully this is correct): http://cl.ly/0V392o061S2O

Here's a screenshot of the issue at very large files: http://cl.ly/3S0q1a3G3P2X

bgrins' test page: http://bgrins.github.io/devtools-demos/debugger/scripts.html

jlongster commented:
Fixed the wide breakpoints, and also fixed the mis-aligned line on my computer. I didn't have "SF UI Text" installed and the line number gutter was trying to use that, but it should be using the monospaced font. Everything looks so much better now!!
